cmd/vgmanager/vgmanager.go

```diff
@@ -192,17 +193,18 @@ func run(cmd *cobra.Command, _ []string, opts *Options) error {
 	}
 
 	if err = (&vgmanager.Reconciler{
-		Client:        mgr.GetClient(),
-		EventRecorder: mgr.GetEventRecorderFor(vgmanager.ControllerName),
-		LVMD:          lvmd.DefaultConfigurator(),
-		Scheme:        mgr.GetScheme(),
-		LSBLK:         lsblk.NewDefaultHostLSBLK(),
-		Wipefs:        wipefs.NewDefaultHostWipefs(),
-		Dmsetup:       dmsetup.NewDefaultHostDmsetup(),
-		LVM:           lvm.NewDefaultHostLVM(),
-		NodeName:      nodeName,
-		Namespace:     operatorNamespace,
-		Filters:       filter.DefaultFilters,
+		Client:          mgr.GetClient(),
+		EventRecorder:   mgr.GetEventRecorderFor(vgmanager.ControllerName),
+		LVMD:            lvmd.DefaultConfigurator(),
+		Scheme:          mgr.GetScheme(),
+		LSBLK:           lsblk.NewDefaultHostLSBLK(),
+		Wipefs:          wipefs.NewDefaultHostWipefs(),
+		Dmsetup:         dmsetup.NewDefaultHostDmsetup(),
+		LVM:             lvm.NewDefaultHostLVM(),
+		NodeName:        nodeName,
+		Namespace:       operatorNamespace,
+		Filters:         filter.DefaultFilters,
+		SymlinkResolver: symlink_resolver.NewWithDefaultResolver(),
```

Comment:
Why do we need this across all reconciles?

Reply:
You can still do the same if you change the injection to the symlink resolution function alone, not the entire resolver object, e.g https://github.com/openshift/lvm-operator/pull/690/files#diff-08a2a7b11748abc371e0feb4f8983dc47562ed33dfbc1861940046a5d14d984aR15 can be injected as a function

```suggestion
		SymlinkFn: filepath.EvalSymlinks,
```

then you can exchange it in the tests